p2panda-rs/src/next/test_utils/db/types/entry.rs

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use crate::next::entry::decode::decode_entry;
+use crate::next::entry::{EncodedEntry, Entry, LogId, SeqNum};
+use crate::next::hash::Hash;
+use crate::next::identity::Author;
+use crate::next::operation::OperationId;
+use crate::next::storage_provider::errors::EntryStorageError;
+use crate::next::storage_provider::traits::AsStorageEntry;
+
+/// A struct which represents an entry and operation pair in storage as a concatenated string.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct StorageEntry {
+    /// Public key of the author.
+    pub author: Author,
+
+    /// Actual Bamboo entry data.
+    pub entry_bytes: EncodedEntry,
+
+    /// Hash of Bamboo entry data.
+    pub entry_hash: Hash,
+
+    /// Used log for this entry.
+    pub log_id: LogId,
+
+    /// Hash of payload data.
+    pub payload_hash: OperationId,
```

Comment:
Think the type is wrong here, this should be `Hash` instead of `OperationId` (the payload hash is not the operation id, its the hash of the entry) :+1: